src/test/java/org/jivesoftware/openfire/auth/JDBCAuthProviderTest.java

```diff
@@ -0,0 +1,79 @@
+package org.jivesoftware.openfire.auth;
+
+import java.util.HashMap;
+import org.junit.Test;
+import static org.junit.Assert.*;
+import org.mindrot.jbcrypt.BCrypt;
+
+public class JDBCAuthProviderTest {
```

Comment:
I was unable to write as many tests as I would like because I had a difficult time bootstrapping the JiveGlobals stuff.  I also didn't feel like making changes to include PowerMock.